pkg/gsm-secrets/execution.go

```diff
@@ -98,19 +104,67 @@ func (a *Actions) CreateServiceAccounts(ctx context.Context, client IAMClient) {
 		secret := a.SecretsToCreate[secretName]
 		secret.Payload = keyData
 		a.SecretsToCreate[secretName] = secret
+	}
+}
+
+// gcpServiceAccountBackoff defines retry behavior for GCP service account operations
+var gcpServiceAccountBackoff = wait.Backoff{
+	Steps:    3,
+	Duration: 8 * time.Second,
+	Factor:   2.0,
+	Jitter:   0.1,
+	Cap:      30 * time.Second,
+}
 
-		logrus.Infof("Created service account: %s", sa.Email)
+// isServiceAccountNotFoundError detects GCP service account "not found" errors indicating eventual consistency
+func isServiceAccountNotFoundError(err error) bool {
+	if err == nil {
+		return false
 	}
+
+	if gcpError, ok := err.(*googleapi.Error); ok {
+		return gcpError.Code == http.StatusNotFound
+	}
+
+	if s, ok := status.FromError(err); ok {
+		return s.Code() == codes.NotFound
+	}
+
+	return false
 }
 
 func GenerateServiceAccountKey(ctx context.Context, client IAMClient, saEmail string, projectID string) ([]byte, error) {
+	return generateServiceAccountKeyWithBackoff(ctx, client, saEmail, projectID, gcpServiceAccountBackoff)
+}
+
+func generateServiceAccountKeyWithBackoff(ctx context.Context, client IAMClient, saEmail string, projectID string, backoff wait.Backoff) ([]byte, error) {
 	name := fmt.Sprintf("%s/serviceAccounts/%s", GetProjectResourceString(projectID), saEmail)
-	key, err := client.CreateServiceAccountKey(ctx, &adminpb.CreateServiceAccountKeyRequest{
-		Name: name,
+	attemptCount := 0
+	var key *adminpb.ServiceAccountKey
+	err := retry.OnError(backoff, isServiceAccountNotFoundError, func() error {
```

Comment:
From my understanding we need to introduce this retry because GCP takes a while to propagate changes across its infra.

Instead of doing this:
```golang
retry.OnError(backoff, isServiceAccountNotFoundError, func() error {
    key, innerErr = client.CreateServiceAccountKey(...)
    if innerErr != nil {
        ...
    }
}
```

Why don't we create the SA and then loop until it shows up?
```golang
client.CreateServiceAccountKey(...)
retry.OnError(backoff, isServiceAccountNotFoundError, func() error {
    // Not sure this call exists but you get the idea
    err = client.GetServiceAccount(...)
    if err != nil {
        ...
    }
}
```

Reply:
I don't understand what's the difference in your suggested approach and what the code's already doing 😅 as it is right now, it tries to create the service account (https://github.com/openshift/ci-tools/pull/4671/files#diff-9909cf7eebaa1d45dbbc42493a83dd1da1e6e4db11e8eed32c3d2f4567945139R148-R150), and if the creation fails, it tries again...